2_process/src/raster_to_catchment_polygons.R

```diff
@@ -80,7 +80,7 @@ raster_to_catchment_polygons <- function(polygon_sf, raster,
   ## Extract and summarize non-descrete raster values in polygons shapefile  
   else{
     message('extracting continuous raster')
-    raster_per_polygon <- terra::extract(raster_crop, vector, fun = raster_summary_fun)
+    raster_per_polygon <- terra::extract(raster_crop, vector, fun = raster_summary_fun, na.rm = na.rm)
```

Comment:
I might have been unclear before. The `na.rm` is one of the `...` arguments in `extract` and it is specific to `raster_summary_fun = sum` (or another function that has na.rm). I think it is likely that we will use a function that has na.rm in it, so this is fine as is. I was suggesting that we could use `...` as an argument in `raster_to_catchment_polygons` and pass that to `extract`. I've not needed to do that before, so I'm not sure if it's as simple as specifying ... in extract, or if you first have to extract what is within the `...`

Reply:
ahh i see, so this would look like this? 

```
raster_to_catchment_polygons <- function(polygon_sf, raster,
                                         categorical_raster = NULL,
                                         raster_summary_fun = NULL,
                                         new_cols_prefix = NULL){
                                         new_cols_prefix = NULL, ...)

```
and 
```
raster_per_polygon <- terra::extract(raster_crop, vector, fun = raster_summary_fun, ...)
```

Reply:
Yes, and then in the in the `tar_target` where the function is called, we'd specify `na.rm = T`. Have you used the `...` in user-defined functions? Does it work like this?

Reply:
I haven't run it. and I've only used `...` when building out merging/appending functions such as `do.call()` or `rbind()` where it can accept an infinite # of input dfs or vectors. I'll run and confirm.

Reply:
@jds485 added this in pipeline now. 
runs the same. so keep the ellipsis here.